pkg/system/system.go

```diff
@@ -171,6 +173,63 @@ func (sc *Controller) registerRouter() {
 	sc.router.HandleFunc(endpointDaemonsUpgrade, sc.upgradeDaemons()).Methods(http.MethodPut)
 	sc.router.HandleFunc(endpointDaemonRecords, sc.getDaemonRecords()).Methods(http.MethodGet)
 	sc.router.HandleFunc(endpointPrefetch, sc.setPrefetchConfiguration()).Methods(http.MethodPut)
+	sc.router.HandleFunc(endpointGetCredential, sc.getCredential()).Methods(http.MethodGet)
+}
+
+func (sc *Controller) getCredential() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var err error
+		var statusCode int
+
+		defer func() {
+			if err != nil {
+				m := newErrorMessage(err.Error())
+				http.Error(w, m.encode(), statusCode)
+			}
+		}()
+
+		vars := mux.Vars(r)
+		id := vars["id"]
+
+		for _, ma := range sc.managers {
+			ma.Lock()
+			d := ma.GetByDaemonID(id)
+
+			if d != nil {
+				backendType, backendConfig := d.Config.StorageBackend()
+				var credentialConfig interface{}
+				switch backendType {
+				case daemonconfig.BackendTypeRegistry:
+					credentialConfig = struct {
+						Auth          string `json:"auth,omitempty"`
+						RegistryToken string `json:"registry_token,omitempty"`
+					}{
+						backendConfig.Auth,
+						backendConfig.RegistryToken,
+					}
+				case daemonconfig.BackendTypeOss:
+					credentialConfig = struct {
+						AccessKeyID     string `json:"access_key_id,omitempty"`
+						AccessKeySecret string `json:"access_key_secret,omitempty"`
+					}{
+						backendConfig.AccessKeyID,
+						backendConfig.AccessKeySecret,
+					}
+				default:
+					err = errdefs.ErrNotFound
+					statusCode = http.StatusNotFound
+					return
+				}
+				jsonResponse(w, credentialConfig)
```

Comment:
Maybe respond to the `backendConfig` directly here? In this way, when we add a new backend, we don't need to worry about handling miss here.